bitcoin/src/address/mod.rs

```diff
@@ -400,8 +418,13 @@ impl Address {
     /// This is the preferred non-witness type address.
     #[inline]
     pub fn p2pkh(pk: &PublicKey, network: Network) -> Address {
-        let payload = Payload::PubkeyHash(pk.pubkey_hash());
-        Address(AddressInner { network, payload }, PhantomData)
+        let hash = pk.pubkey_hash();
+        Address::_p2pkh(hash, network)
+    }
+
+    fn _p2pkh(hash: PubkeyHash, network: Network) -> Address {
+        let prefix = LegacyP2pkhPrefix::from_network(network);
+        Self(AddressInner::P2pkh { hash, prefix }, PhantomData)
```

Comment:
Might be nicer to still have have `fn new(inner: AddressInner) -> Self` so that we can avoid `PhantomData`.

Reply:
I had a go but I think it the code is marginally easier to read with the explicit `PhantomData` instead of 
```rust
Self::new(AddressInner::P2pkh { hash, prefix })
```
Saves one thinking "what does new do"?